### PR TITLE
Fixes broken link of csi-hostpath-plugin.yaml

### LINKED ISSUE #369: Need to locate the correct path of `csi-hostpath-plugin.yaml`
## Issue of Broken Link
The issue I have observed is that when we visit [ this-page]( https://github.com/kubernetes-csi/csi-driver-host-path/blob/master/docs/deploy-1.17-and-later.md) and go to the `Confirm Hostpath driver works` this section and click on  [here](https://github.com/kubernetes-csi/csi-driver-host-path/blob/master/deploy/kubernetes-1.17/hostpath/csi-hostpath-plugin.yaml)  rather than getting container hostpath statefulset plugin yml configuration we got this unwanted page. because we don't have the directory kubernetes-1.17 present inside the deploy directory.

## Solution 
We can provide the correct path of `csi-hostpath-plugin.yaml` which is present inside the `kubernetes-1.22-test` directory.

/assign 
/kind bug

### PR DESCRIPTION
**Pull request by amolmote**
Signed-off-by: amolmote <amolmote201@gmail.com>

### CODE REVIEW

File: docs/deploy-1.17-and-later.md [hunk 1]
```diff
@@ -219,7 +219,7 @@ Events:
 ```
 
 ## Confirm Hostpath driver works
-The Hostpath driver is configured to create new volumes under `/csi-data-dir` inside the hostpath container that is specified in the plugin StatefulSet found [here](../deploy/kubernetes-1.17/hostpath/csi-hostpath-plugin.yaml).  This path persist as long as the StatefulSet pod is up and running.
+The Hostpath driver is configured to create new volumes under `/csi-data-dir` inside the hostpath container that is specified in the plugin StatefulSet found [here](https://github.com/kubernetes-csi/csi-driver-host-path/blob/master/deploy/kubernetes-1.22-test/hostpath/csi-hostpath-plugin.yaml).  This path persist as long as the StatefulSet pod is up and running.
```
Comment (pohly) [inline docs/deploy-1.17-and-later.md:222]: Can you keep this as a relative link?

Comment (amolmote) [reply docs/deploy-1.17-and-later.md:222]: sure will try @pohly

Comment (amolmote) [inline docs/deploy-1.17-and-later.md:222]:
```suggestion
The Hostpath driver is configured to create new volumes under `/csi-data-dir` inside the hostpath container that is specified in the plugin StatefulSet found [here](../deploy/kubernetes-1.22-test/hostpath/csi-hostpath-plugin.yaml).  This path persist as long as the StatefulSet pod is up and running.
```

Comment (amolmote) [reply docs/deploy-1.17-and-later.md:222]: @pohly, Do you want like this?

Comment (pohly) [reply docs/deploy-1.17-and-later.md:222]: Yes. Please squash into one commit.

Comment (amolmote) [reply docs/deploy-1.17-and-later.md:222]: yes

Comment (amolmote) [reply docs/deploy-1.17-and-later.md]: squash done

**Review by pohly — APPROVED** [review]
/lgtm

**Comment by amolmote** [comment]
this pr is waiting for ok-to-test lable

**Pull request merged**